pkg/subsystem/linux/subsystems.go

```diff
@@ -135,14 +137,37 @@ func (ctx *linuxCtx) groupByRules() ([]*subsystem.Subsystem, error) {
 	return ret, nil
 }
 
-func (ctx *linuxCtx) applyExtraRules(list []*subsystem.Subsystem) {
+func (ctx *linuxCtx) applyExtraRules(list []*subsystem.Subsystem) error {
 	if ctx.extraRules == nil {
-		return
+		return nil
 	}
+	perName := map[string]*subsystem.Subsystem{}
 	for _, entry := range list {
 		entry.Syscalls = ctx.extraRules.subsystemCalls[entry.Name]
 		_, entry.NoReminders = ctx.extraRules.noReminders[entry.Name]
+		perName[entry.Name] = entry
+	}
+	for from, toNames := range ctx.extraRules.addParents {
+		item := perName[from]
+		if item == nil {
+			return fmt.Errorf("unknown subsystem: %q", from)
+		}
+		exists := map[string]bool{}
+		for _, p := range item.Parents {
+			exists[p.Name] = true
+		}
+		for _, toName := range toNames {
+			if exists[toName] {
+				continue
+			}
+			if perName[toName] == nil {
+				return fmt.Errorf("unknown parent subsystem: %q", toName)
+			}
+			item.Parents = append(item.Parents, perName[toName])
```

Comment:
Test it please.
This function is quite hard to read and understand. Please comment it.

Reply:
Added a test.